environments/php7/server.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+require __DIR__ . '/vendor/autoload.php';
+
+use Monolog\Handler\StreamHandler;
+use Monolog\Logger;
+use PhpParser\ParserFactory;
+use Psr\Http\Message\ServerRequestInterface;
+use React\Http\Server;
+use RingCentral\Psr7\Response;
+
+const V1_CODEPATH = '/userfunc/user';
+const V1_USER_FUNCTION = 'handler';
+const HANDLER_DIVIDER = '::';
+
+$codePath = null;
+$userFunction = null;
+$logger = new Logger("Function");
+$logger->pushHandler(new StreamHandler('php://stdout', Logger::DEBUG));
+
+
+$loop = React\EventLoop\Factory::create();
+
+$server = new Server(function (ServerRequestInterface $request) use (&$codePath, &$userFunction, $logger) {
+    $path = $request->getUri()->getPath();
+    $method = $request->getMethod();
+
+    if ('/specialize' === $path && 'POST' === $method) {
+        $codePath = V1_CODEPATH;
+        $userFunction = V1_USER_FUNCTION;
+
+        return new Response(201);
+    }
+
+    if ('/v2/specialize' === $path && 'POST' === $method) {
+        $body = json_decode($request->getBody()->getContents(), true);
+        $filepath = $body['filepath'];
+        list ($moduleName, $userFunction) = explode(HANDLER_DIVIDER, $body['functionName']);
+        if (true === is_dir($filepath)) {
+            $codePath = $filepath . DIRECTORY_SEPARATOR . $moduleName;
+
+        } else {
+            $codePath = $filepath;
+        }
+
+        return new Response(201);
+    }
+    if ('/' === $path) {
+        if (null === $codePath) {
+            $logger->error("$codePath not found");
+            return new Response(500, [], 'Generic container: no requests supported');
+        }
+
+        ob_start();
+
+        if (!file_exists($codePath)) {
+            $logger->error("$codePath not found");
+            return new Response(500, [], "$codePath not found");
+        }
+
+        $parser = (new ParserFactory)->create(ParserFactory::PREFER_PHP7);
+        try {
+            $parser->parse(file_get_contents($codePath));
+        } catch (Throwable $throwable) {
+            $logger->error($codePath . ' - ' . $throwable->getMessage());
+            return new Response(500, [], $codePath . ' - ' . $throwable->getMessage());
+        }
+
+        require $codePath;
```

Comment:
I guess it's the root cause because server requires user function twice.

Reply:
maybe add a flag to not require again if we already did that before. Or, any better suggestion?

Reply:
After moving `require` to specialize stage it works perfectly. Once this problem is addressed then we can merge it! 💯

Reply:
@life1347 Fixed! I changed `require` to `require_once`. This prevents that behavior :)